federation/pkg/dnsprovider/providers/coredns/coredns.go

```diff
@@ -48,11 +57,36 @@ func init() {
 	})
 }
 
+func newTransportForETCD2(certFile, keyFile, caFile string) (*http.Transport, error) {
```

Comment:
for etcdv3 TLSConfig is a part of etcd.Config - https://github.com/coreos/etcd/blob/master/clientv3/config.go#L37